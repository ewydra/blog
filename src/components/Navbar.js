import React from "react";
import { Link } from "react-router-dom";
import { isLoggedIn } from "../selectors/users";
import LogoutButton from "./LogoutButton";
import { useSelector } from "react-redux";

const Navbar = () => {
  const isAuthenticated = useSelector(isLoggedIn);

  return (
  <nav className="navbar">
    <div className="menu">
      <Link className="navbar__link" to={{ pathname: "/" }}>
        Home page
      </Link>
      {isAuthenticated && (
        <Link className="navbar__link" to={{pathname: "/article/new" }}>
          New article
        </Link>
      )}
    </div>
    <div className="menu">
      {isAuthenticated ? (
        <LogoutButton />
      ) : (
        <>
          <Link className="navbar__link" to={{ pathname: "/login" }}>
            Login
          </Link>
          <Link className="navbar__link" to={{ pathname: "/register" }}>
            Register
          </Link>
        </>
      )}
    </div>
  </nav>
)};

export default Navbar;
