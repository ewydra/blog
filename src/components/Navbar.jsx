import React from 'react';
import { Link } from "react-router-dom";

const Navbar = () => (
  <nav className="Navbar" >
    <Link className="Navbar__link" to="/">Home page</Link>
  </nav>
);

export default Navbar;
