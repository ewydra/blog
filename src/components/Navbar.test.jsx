import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('should render correctly', () => {
    const component = shallow(<Navbar />);
    expect(component).toMatchSnapshot();
  });

  it('should include links to home page and new article page', () => {                                       
    const component = mount(<MemoryRouter><Navbar /></MemoryRouter>);
    expect(component).toContainReact(<Link className="Navbar__link" to="/">Home page</Link>);
    expect(component).toContainReact(<Link className="Navbar__link" to="/article/new">New article</Link>);
  });
});
