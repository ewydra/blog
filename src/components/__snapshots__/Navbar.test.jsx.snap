// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<nav
  className="Navbar"
>
  <Link
    className="Navbar__link"
    to="/"
  >
    Home page
  </Link>
  <Link
    className="Navbar__link"
    to="/article/new"
  >
    New article
  </Link>
</nav>
`;
